src/legacy/ui/public/agg_types/controls/string.tsx

```diff
@@ -41,7 +41,11 @@ function StringSelect({ agg = {}, aggParam = {}, onParamsChange }: StringSelectP
 
   return (
     <EuiFormRow label={aggParam.displayName || aggParam.name} className="form-group">
-      <EuiFieldText id={`visEditorStringInput${agg.id}${aggParam.name}`} onChange={onChange} />
+      <EuiFieldText
+        id={`visEditorStringInput${agg.id}${aggParam.name}`}
+        name={`visEditorStringInput${agg.id}${aggParam.name}`}
```

Comment:
It would be nice to add `data-test-subj` here and re-use it in a functional test.

Reply:
thank you for the comment. I added `data-test-subj` and updated `test/functional/page_objects/visualize_page.js`.